--- x_interface.c.orig	2017-03-08 10:32:08.000000000 -0700
+++ x_interface.c	2025-09-05 12:57:05.879780751 -0600
@@ -438,8 +438,8 @@
 void 	do_set_dataedit_place  (Widget w, XButtonEvent *e, String *p, Cardinal *n );
 void 	do_set_min_from_curdata(Widget w, XButtonEvent *e, String *p, Cardinal *n );
 void 	do_set_max_from_curdata(Widget w, XButtonEvent *e, String *p, Cardinal *n );
-void 	expose_ccontour();
-void 	expose_colorbar();
+void 	expose_ccontour(Widget w, XtPointer client_data, XExposeEvent *event, Boolean *continue_to_dispatch );
+void 	expose_colorbar(Widget w, XtPointer client_data, XExposeEvent *event, Boolean *continue_to_dispatch );
 
 void 	testf(Widget w, XButtonEvent *e, String *p, Cardinal *n );
 
@@ -613,7 +613,7 @@
 /*************************************************************************************************/
 void x_initialize()
 {
-	void	  check_app_res();
+	void	  check_app_res(AppDataPtr ad);
 	void 	  track_pointer();
 
 	static XtActionsRec new_actions[] = {
