--- plugins/uefi-capsule/make-images.py.orig	2024-08-21 11:52:17.000000000 -0600
+++ plugins/uefi-capsule/make-images.py	2025-09-06 16:51:43.898005352 -0600
@@ -146,8 +146,17 @@
                 items = Pango.itemize(pctx, label_translated, 0, length, attrs, None)
                 if not items:
                     continue
-                gs = Pango.GlyphString()
-                Pango.shape(label_translated, length, items[0].analysis, gs)
+                # https://github.com/fwupd/fwupd/pull/8588/files
+                try:
+                    # urgh, https://gitlab.gnome.org/GNOME/pango/-/merge_requests/829
+                    # -- if we depend on Pango >= 1.56.2 we can drop the fallback
+                    #
+                    # ...or if we depend on python3-gobject >= 3.51.0 we can use the nicer:
+                    #    len(inspect.signature(Pango.shape).parameters)
+                    gs = Pango.shape(label_translated, length, items[0].analysis)
+                except TypeError:
+                    gs = Pango.GlyphString()
+                    Pango.shape(label_translated, length, items[0].analysis, gs)
                 del img, cctx, pctx, layout
 
                 def find_size(fs, f, data):
